pages/cancel/[uid].tsx

```diff
@@ -1,71 +1,37 @@
 import { CalendarIcon, XIcon } from "@heroicons/react/solid";
 import dayjs from "dayjs";
-import utc from "dayjs/plugin/utc";
-import { getSession } from "next-auth/client";
+import { GetServerSidePropsContext } from "next";
 import { useRouter } from "next/router";
 import { useState } from "react";
 
+import { asStringOrUndefined } from "@lib/asStringOrNull";
+import { getSession } from "@lib/auth";
 import { useLocale } from "@lib/hooks/useLocale";
 import prisma from "@lib/prisma";
 import { collectPageParameters, telemetryEventTypes, useTelemetry } from "@lib/telemetry";
+import { inferSSRProps } from "@lib/types/inferSSRProps";
 
 import CustomBranding from "@components/CustomBranding";
 import { HeadSeo } from "@components/seo/head-seo";
 import { Button } from "@components/ui/Button";
 
-dayjs.extend(utc);
-
-export default function Type(props) {
+export default function Type(props: inferSSRProps<typeof getServerSideProps>) {
   const { t } = useLocale();
   // Get router variables
   const router = useRouter();
   const { uid } = router.query;
-
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const [is24h, setIs24h] = useState(false);
+  const [is24h] = useState(false);
   const [loading, setLoading] = useState(false);
-  const [error, setError] = useState(props.booking ? null : t("booking_already_cancelled"));
+  const [error, setError] = useState<string | null>(props.booking ? null : t("booking_already_cancelled"));
   const telemetry = useTelemetry();
 
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
-  const cancellationHandler = async (event) => {
-    setLoading(true);
-
-    const payload = {
-      uid: uid,
-    };
-
-    telemetry.withJitsu((jitsu) =>
-      jitsu.track(telemetryEventTypes.bookingCancelled, collectPageParameters())
-    );
-
-    const res = await fetch("/api/cancel", {
-      body: JSON.stringify(payload),
-      headers: {
-        "Content-Type": "application/json",
-      },
-      method: "DELETE",
-    });
-
-    if (res.status >= 200 && res.status < 300) {
-      await router.push(
-        `/cancel/success?name=${props.profile.name}&title=${props.booking.title}&eventPage=${
-          props.profile.slug
-        }&team=${props.booking.eventType.team ? 1 : 0}`
-      );
-    } else {
-      setLoading(false);
-      setError(`${t("error_with_status_code_occured", { status: res.status })} ${t("please_try_again")}`);
-    }
-  };
```

Comment:
Moved inline to take advantage of [contextual types](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/forms_and_events/)